File: src/test/java/no/echokarriere/backend/DatabaseExtension.java
package no.echokarriere.backend;

import org.flywaydb.core.Flyway;
import org.flywaydb.core.api.configuration.FluentConfiguration;
import org.junit.jupiter.api.extension.AfterAllCallback;
import org.junit.jupiter.api.extension.BeforeAllCallback;
import org.junit.jupiter.api.extension.ExtensionContext;

public class DatabaseExtension implements BeforeAllCallback, AfterAllCallback {
    private Flyway flyway() {
        var config = new FluentConfiguration()
                .dataSource("jdbc:postgresql://localhost:32789/test_db", "karriere", "password");
        return new Flyway(config);
    }

    @Override
    public void afterAll(ExtensionContext context) {
        flyway().clean();
    }

    @Override
    public void beforeAll(ExtensionContext context) {
        flyway().migrate();
    }
}

File: src/test/kotlin/utils/DatabaseExtension.kt
package no.echokarriere.utils

import org.flywaydb.core.Flyway
import org.junit.jupiter.api.extension.AfterAllCallback
import org.junit.jupiter.api.extension.BeforeAllCallback
import org.junit.jupiter.api.extension.ExtensionContext

class DatabaseExtension : BeforeAllCallback, AfterAllCallback {
    private val flyway = Flyway.configure()
        .locations("filesystem:src/main/resources/db/migrations")
        .dataSource("jdbc:postgresql://localhost:32782/echokarriere?loggerLevel=OFF", "karriere", "password")
        .load()

    override fun beforeAll(context: ExtensionContext?) {
        flyway.migrate()
    }

    override fun afterAll(context: ExtensionContext?) {
        flyway.clean()
    }
}
